addons/website_mail/controllers/main.py

```diff
@@ -76,55 +75,50 @@ class WebsiteMail(http.Controller):
-        _object = request.registry[object]
+        _object = request.env[object]
 
         # search partner_id
         public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_ids = [user_obj.browse(cr, uid, uid, context).partner_id.id]
+        if request.env.uid != public_id:
+            partner_ids = [request.env.user.partner_id.id]
         else:
             # mail_thread method
-            partner_ids = _object._find_partner_from_emails(
-                cr, SUPERUSER_ID, [_id], [email], context=context, check_followers=True)
+            partner_ids = _object.sudo()._find_partner_from_emails([email], check_followers=True)
             if not partner_ids or not partner_ids[0]:
                 name = email.split('@')[0]
-                partner_ids = [partner_obj.create(cr, SUPERUSER_ID, {'name': name, 'email': email}, context=context)]
-
+                partner_ids = [Partner.create({'name': email, 'email': email}).id]
         # add or remove follower
+        object_record = _object.sudo().browse(_id)
         if _message_is_follower:
-            _object.check_access_rule(cr, uid, [_id], 'read', context)
-            _object.message_unsubscribe(cr, SUPERUSER_ID, [_id], partner_ids, context=context)
+            object_record.check_access_rule('read')
+            object_record.message_unsubscribe(partner_ids)
             return False
         else:
-            _object.check_access_rule(cr, uid, [_id], 'read', context)
+            object_record.check_access_rule('read')
             # add partner to session
             request.session['partner_id'] = partner_ids[0]
-            _object.message_subscribe(cr, SUPERUSER_ID, [_id], partner_ids, context=context)
+            object_record.message_subscribe(partner_ids)
             return True
 
     @http.route(['/website_mail/is_follower'], type='json', auth="public", website=True)
     def call(self, model, id, **post):
         id = int(id)
-        cr, uid, context = request.cr, request.uid, request.context
+        uid = request.env.uid
 
-        partner_obj = request.registry.get('res.partner')
-        users_obj = request.registry.get('res.users')
-        obj = request.registry.get(model)
+        Partner = request.env['res.partner']
+        Obj = request.env[model]
 
-        partner_id = None
+        partner_record = None
         public_id = request.website.user_id.id
         if uid != public_id:
-            partner_id = users_obj.browse(cr, SUPERUSER_ID, uid, context).partner_id
+            partner_record = request.env.user.partner_id
         elif request.session.get('partner_id'):
-            partner_id = partner_obj.browse(cr, SUPERUSER_ID, request.session.get('partner_id'), context)
-        email = partner_id and partner_id.email or ""
+            partner_record = Partner.sudo().browse(request.session.get('partner_id'))
+        email = partner_record and partner_record.email or ""
```

Comment:
remove this statement and assign 'partner_record.email' in values dictionary below.